org.alloytools.api/src/main/java/org/alloytools/alloy/dto/SolutionDTO.java

```diff
@@ -0,0 +1,98 @@
+package org.alloytools.alloy.dto;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A Data Transfer Object (DTO) representing a specific Alloy solution.
+ *
+ * This class provides a structured way to encapsulate and transfer data related
+ * to a particular solution generated from an Alloy model.
+ */
+public class SolutionDTO {
+
+    /**
+     * Indicates whether the solution is satisfiable, i.e., whether a valid instance
+     * of the model exists that satisfies all constraints.
+     */
+    public boolean               satisfiable;
```

Comment:
Isn't this true IFF the length of the `instances` list is greater than 0?

Reply:
I guess you're right. But the JSON tends to be used in Javascript and it is more readable to check the flag than the list of instances.